File: src/Entity/Centre.php
<?php

namespace App\Entity;

use App\Repository\CentreRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * @ORM\Entity(repositoryClass=CentreRepository::class)
 */
class Centre
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;


    /**
     * @Assert\NotBlank(message="nom est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $nom;

    /**
     * @Assert\NotBlank(message="telephone est obligatoire")
     * @ORM\Column(type="string", length=255)
     */
    private $telephone;

    /**
     * @Assert\NotBlank(message="Email est obligatoire")
     * @Assert\Email(message = " email '{{ value }}' n'est pas valide.")
     * @ORM\Column(type="text")
     */
    private $email;

    /**
     * @ORM\Column(type="time")
     */
    private $horaire;

    /**
     * @Assert\NotBlank(message="description est obligatoire")
     * @ORM\Column(type="text")
     */
    private $description;
    /**
     * @ORM\Column(type="string", length=255)
     */
    private $localisation;
    /**
     * @var \Doctrine\Common\Collections\ArrayCollection
     * @ORM\ManyToMany(targetEntity=CategorieCentre::class, inversedBy="centres")
     * @ORM\JoinTable(name="centre_categorie_centre")
     */
    private $categorie;

    /**
     * @ORM\OneToMany(targetEntity=CentreLike::class, mappedBy="centre", orphanRemoval=true)
     */
    private $yes;

    public function __construct()
    {
        $this->categorie = new ArrayCollection();
        $this->yes = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }


    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getTelephone(): ?string
    {
        return $this->telephone;
    }

    public function setTelephone(string $telephone): self
    {
        $this->telephone = $telephone;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getHoraire(): ?\DateTimeInterface
    {
        return $this->horaire;
    }

    public function setHoraire(\DateTimeInterface $horaire): self
    {
        $this->horaire = $horaire;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    /**
     * @return Collection|CategorieCentre[]
     */
    public function getCategorie(): Collection
    {
        return $this->categorie;
    }

    public function addCategorie(CategorieCentre $categorie): self
    {
        if (!$this->categorie->contains($categorie)) {
            $this->categorie[] = $categorie;
        }

        return $this;
    }

    public function removeCategorie(CategorieCentre $categorie): self
    {
        $this->categorie->removeElement($categorie);

        return $this;
    }



    public function getLocalisation(): ?string
    {
        return $this->localisation;
    }

    public function setLocalisation(string $localisation): self
    {
        $this->localisation = $localisation;

        return $this;
    }

    /**
     * @return Collection|CentreLike[]
     */
    public function getYes(): Collection
    {
        return $this->yes;
    }

    public function addYe(CentreLike $ye): self
    {
        if (!$this->yes->contains($ye)) {
            $this->yes[] = $ye;
            $ye->setCentre($this);
        }

        return $this;
    }

    public function removeYe(CentreLike $ye): self
    {
        if ($this->yes->removeElement($ye)) {
            // set the owning side to null (unless already changed)
            if ($ye->getCentre() === $this) {
                $ye->setCentre(null);
            }
        }

        return $this;
    }
    /**
     * @param User $user
     * @return bool
     */
    public function isLikedByUser(User $user){
        foreach ($this->yes as $likes){
            if ($likes->getUser() === $user){
                return true;
            }
        }
        return false;

    }
}

File: sprint java/Entities/Centre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;





/**
 *
 * @author DEMNI Anis
 */


public class Centre {
   
     
 
  private String nom,email,horaire,description,localisation;
  
  private int id,telephone ;
  
    public Centre() {
    }

    public Centre(int id , String nom, int telephone, String email, String horaire, String description, String localisation) {
        this.id=id;
        this.nom=nom;
        this.telephone=telephone;
        this.email=email;
        this.horaire=horaire;
        this.description=description;
        this.localisation=localisation;
       
      
    }

    public Centre(String nom,  int telephone, String email, String horaire, String description,String localisation) {
        this.nom = nom;
        this.email = email;
        this.horaire = horaire;
        this.description = description;
        this.localisation = localisation;
        this.telephone = telephone;
    }

    


    
  

    


   

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getHoraire() {
        return horaire;
    }

    public void setHoraire(String horaire) {
        this.horaire = horaire;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getLocalisation() {
        return localisation;
    }

    public void setLocalisation(String localisation) {
        this.localisation = localisation;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getTelephone() {
        return telephone;
    }

    public void setTelephone(int telephone) {
        this.telephone = telephone;
    }


    
    @Override
    public String toString() {
        return "Centre{" + "nom=" + nom + ", telephone=" + telephone + ", email=" + email + ", horaire=" + horaire + ", description=" + description + ", localisation=" + localisation+'}';
    }

    
}
